source/libip6string/ip6tos.c

```diff
@@ -92,4 +95,27 @@ void ip6tos(const void *ip6addr, char *p)
         }
     }
     *p++ = '\0';
+
+    // Return length of generated string, excluding the terminating null-character
+    return p - p_orig - 1;
+}
+
+uint8_t ip6_prefix_tos(const uint8_t *prefix, uint8_t prefix_len, char *p)
+{
+    char *wptr = p;
+    uint8_t addr[16] = {0};
+
+    if (((prefix_len != 0) && (prefix == NULL )) || prefix_len > 128) {
```

Comment:
Stray space after NULL, and for me this is past the "too much bracketing" threshold.

Isn't this clearer?

    if ((prefix_len != 0 && prefix == NULL) || prefix_len > 128)

Not actually sure if it's worth being defensive here, rather than in the outer layer. ip6tos isn't being defensive, but then I guess there's less to get wrong there.